sklearn/multioutput.py

```diff
@@ -359,7 +359,14 @@ def fit(self, X, Y, sample_weight=None):
         self.classes_ = [estimator.classes_ for estimator in self.estimators_]
         return self
 
-    def predict_proba(self, X):
+    def _check_proba(self):
```

Comment:
Not sure it's worth abstracting the method away since you're only using it in `predict_proba`. Same for `_predict_proba`

Reply:
I thought it'd be more consistent stylistically, as [SVC does something similar](https://github.com/scikit-learn/scikit-learn/blob/d94e2a2f0067cc8539672814abc0ec394e21656d/sklearn/svm/base.py#L581), which is [what was suggested](https://github.com/scikit-learn/scikit-learn/pull/12222#issuecomment-482647023) in the related PR.

I can put everything in `predict_proba` if that's better.

Reply:
It's used in more than one place in SVC so that makes sense, but not here. So I'd just put everything inside `predict_proba`. We can refactor when/if the time comes ;)

Reply:
But we'd still need `_predict_proba` since you'd need can't pass in `X` when doing `hasattr(clf, "predict_proba")`

Reply:
You don't need to pass X (you don't pass self either):


```py
In [1]: class C:
   ...:     def method(self, a, b, c):
   ...:         pass
   ...:     

In [2]: hasattr(C(), 'method')
Out[2]: True
```

Or am I missing your point?

Reply:
When the function signature is `def predict_proba(self, X)`, I get
```TypeError: predict_proba() missing 1 required positional argument: 'X'```
for `assert not hasattr(multi_target_linear, "predict_proba")`.

Other tests like ` y_result = clf.predict_proba(X)` fails with 
```TypeError: predict_proba() missing 1 required positional argument: 'X'```

How do I get around that?

Reply:
When decorated with @property, the function can only take `self` as a parameter, since the function is then a getter. But the getter can return a function with whatever parameters

Reply:
Right, which is why I thought we needed the `_predict_proba` method. @NicolasHug thoughts? 
If not, let me know but I think this current implementation is ready to merge.

Reply:
Sorry, I was forgetting that you can't pass additional arguments to the property. This is good